### PR TITLE
Backported device renaming fixes

### PR DESCRIPTION
**Pull request by mchf**
bnc#900383

### CODE REVIEW

File: library/network/src/NetworkInterfaces.ycp [hunk 1]
```diff
@@ -226,25 +349,64 @@ global string GetTypeFromIfcfg(ifcfg_t ifcfg) {
...
+    {
 	ifcfg_t ifcfg = confs[dev]:$[];
-	string ifcfg_type = GetTypeFromIfcfg(ifcfg);
-	if (ifcfg_type != nil)
-	  type = ifcfg_type;
-      });
-//y2internal("original %1: %2", dev, type);
+        
+        if( !IsEmpty( ifcfg) )
+            type = GetTypeFromIfcfgOrName( dev, ifcfg);
```
Comment (jreidinger) [inline library/network/src/NetworkInterfaces.ycp]: so type is overwritten everytime and last win?

**Comment by jreidinger** [comment]
just one note, otherwise it looks good

File: library/network/src/NetworkInterfaces.ycp (hunk 1 above)
Comment (mchf) [inline library/network/src/NetworkInterfaces.ycp]: Seems so ... but no. This code is consequence of complexity in structure where cached configuration is stored. It in fact only tries to find cached sysconfig configuration for particular device name.

**Comment by mvidner** [comment]
Note for myself: In case of backports, I like to include a reference to the original commits. (It helps me do a diff of diffs, to check for possible git accidents).
In this case, it seems it is a port of #90 and #91, with the relevant diff being "git diff 3ec0c3a1357b6eccade6b863dbb8af02733837e3..39d3bd05d4234fa63c6c1b0c4a4b1bdc6be9da0b"

File: package/yast2.changes [hunk 2]
```diff
@@ -1,4 +1,17 @@
 -------------------------------------------------------------------
+Wed Nov  5 09:33:08 UTC 2014 - mfilka@suse.com
+
+- bnc#900383
+  - ported several fixes for device renaming
```
Comment (mvidner) [inline package/yast2.changes]: So this item is a summary of the remaining items, isn't it? Then it is confusing to list it on the same level.

File: package/yast2.changes [hunk 3]
```diff
@@ -1,4 +1,17 @@
...
+  - added net device type detection based on sysfs
+  - fixed type detection workflow
+  - fixed device type detection when commiting new device into
+    NetworkInterfaces' cache.
```
Comment (mvidner) [inline package/yast2.changes]: Do include the original Bugzilla reference, [bnc#809053](https://bugzilla.suse.com/show_bug.cgi?id=809053).

File: package/yast2.changes [hunk 4]
```diff
@@ -1,4 +1,17 @@
...
+  - changes API for device type detection - incompatible to previous
+    versions
```
Comment (mvidner) [inline package/yast2.changes]: This is the biggest potential problem. How do you ensure that yast2-network, which contains 1 reference to the old API, does not break?

Comment (mchf) [inline package/yast2.changes]: via bugzilla or I can update ```Conflicts``` directive in spec file. The second one seems better to me. Do you agree with the proposal or you have an other idea?

**Comment by mvidner** [comment]
OK, a "Conflicts: yast2-network < ..." here sounds good. Please make a PR for network so that we can check that they go well together.

File: yast2.spec.in [hunk 5]
```diff
@@ -42,7 +42,8 @@ Conflicts:      yast2-inetd <= 2.13.4
 Conflicts:      yast2-tune < 2.15.6
 Obsoletes:      yast2-mail-aliases <= 2.14.0
 Conflicts:      yast2-storage < 2.16.4
-Conflicts:      yast2-network < 2.16.6
+# older yast2-network uses old device type detection API
+Conflicts:      yast2-network < 2.17.135.1
```
Comment (mvidner) [inline yast2.spec.in]: 2.17.199.1

**Comment by mvidner** [comment]
Please fix the version in the conflict, then it LGTM.

**Comment by mchf** [comment]
I've retested one more time to be sure that complicated integration didn't break anything. Seems ok.

Thanks all for reviews.
@mvidner: many thanks for your perfect job.

**Pull request merged**